### ISSUE TITLE
/shop performance: product query improvements

### ISSUE THREAD

**Issue by Matt-Yorkley** [openfoodfoundation/openfoodnetwork#3853]
## Description

Loading enterprise/shop invokes n+1 queries. This results in a huge server load and impacts on performance.

This issue is capturing work to be done within the enterprise/shop performance feature.

## Expected Behavior

Shopfronts always load successfully.

## Actual Behaviour

Occasionally shopfronts fail to load and timeout

## Steps to Reproduce

1. Load enterprise/shop
2. Take a look on Datadog for number of queries run
3. See that we do around 11 queries for every product in the OC, so with 500 products, we make ~5500 queries. 

![Screenshot from 2019-08-02 15-57-40](https://user-images.githubusercontent.com/762088/62375317-5038a480-b53e-11e9-8227-b2654bc7d8f4.png)

## Context

This issue gives an overview of inflated querying and places to address them within enterprise/shop epic #3860 

Each of the possible fixes below should be addressed in a separate PR as much as possible such that they can be tested independently.

## Possible Fix

The performance of the products query for a shop is really bad. It starts here in the shop controller, but touches many different files:

https://github.com/openfoodfoundation/openfoodnetwork/blob/59d4e3d9e2192524d807b8dbc4271b427adea1ac/app/controllers/shop_controller.rb#L12-L24

https://github.com/openfoodfoundation/openfoodnetwork/blob/59d4e3d9e2192524d807b8dbc4271b427adea1ac/lib/open_food_network/cached_products_renderer.rb#L29-L39

https://github.com/openfoodfoundation/openfoodnetwork/blob/59d4e3d9e2192524d807b8dbc4271b427adea1ac/lib/open_food_network/products_renderer.rb#L34-L45

https://github.com/openfoodfoundation/openfoodnetwork/blob/59d4e3d9e2192524d807b8dbc4271b427adea1ac/app/services/order_cycle_distributed_products.rb#L17-L27

**Comment by Matt-Yorkley** [comment]
With ~500 products in an OC:

![products](https://user-images.githubusercontent.com/9029026/57839579-e7535300-77be-11e9-93a8-a63b22d496c7.png)

**Comment by lin-d-hop** [comment]
First round spike time boxed 1hr max.

Delete dead code (last one) duplicated in #4251

**Comment by sauloperez** [comment]
Lots of improvements have been made in https://github.com/openfoodfoundation/openfoodnetwork/pull/4334 touching all the remaining areas suggested here so we can't tell what's its impact until it reaches production and we gather metrics for at least a week. However, I don't see any obvious room for improvement other than what we did already.

Because of that I focused on `EnterprisesController#shop` (/<enterprise_permalink>/shop) as there's room there.
